server/server.go

```diff
@@ -285,6 +285,16 @@ func (s *Server) Run() error {
 	return nil
 }
 
+// SetUpTestMode for testing
+func (s *Server) SetUpTestMode() {
```

Comment:
Can we do it in `server.CreateServer`? It looks verbose to call it in every tests...